src/functions/retryProcessing/application/__tests__/RetryProcessor.int.ts

```diff
@@ -182,5 +193,127 @@ describe('RetryProcessor database test', () => {
         });
     });
   };
+  const truncateTestresults = (): Promise<number> => {
+    return new Promise((resolve, reject) => {
+      db.query(
+          `
+          TRUNCATE TABLE TEST_RESULT
+          `,
+          [],
+          (err, results, fields) => {
+            if (err) {
+              reject(err);
+            }
+            resolve(1);
+          });
+    });
+  };
+  const truncateUploadQueue = (): Promise<number> => {
+    return new Promise((resolve, reject) => {
+      db.query(
+          `
+          TRUNCATE TABLE UPLOAD_QUEUE
+          `,
+          [],
+          (err, results, fields) => {
+            if (err) {
+              reject(err);
+            }
+            resolve(1);
+          });
+    });
+  };
+  const createRetryScenario = (appRef: number, resultStatus: string,
+                               dateTime: string,
+                               tarsStatus: string,
+                               tarsRetryCount: number,
+                               rsisStatus: string,
+                               rsisRetryCount: number,
+                               notifyStatus: string,
+                               notifyRetryCount: number): Promise<number> => {
+    return new Promise((resolve, reject) => {
+      const formattedQuery = db.format(
+            `CALL sp_create_retry_process_scenario(?,?,?,?,?,?,?,?,?)`, [
+              appRef, resultStatus, dateTime, tarsStatus, tarsRetryCount,
+              rsisStatus, rsisRetryCount, notifyStatus, notifyRetryCount]);
+      db.query(
```

Comment:
I like this, probably worth adding one for calling the autosave procedure too or editing this one to be able to use an autosave flag as the only thing changing will be the procedure name